indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -160,48 +164,55 @@ public String setup(TaskToolbox toolbox) throws Exception
   @Override
   public final TaskStatus run(TaskToolbox taskToolbox) throws Exception
   {
-    boolean failure = false;
+    TaskStatus taskStatus = TaskStatus.running(getId());
     try {
       String errorMessage = setup(taskToolbox);
       if (org.apache.commons.lang3.StringUtils.isNotBlank(errorMessage)) {
         return TaskStatus.failure(getId(), errorMessage);
       }
-      TaskStatus taskStatus = runTask(taskToolbox);
-      if (taskStatus.isFailure()) {
-        failure = true;
-      }
+      taskStatus = runTask(taskToolbox);
       return taskStatus;
     }
     catch (Exception e) {
-      failure = true;
+      taskStatus = TaskStatus.failure(getId(), e.toString());
       throw e;
     }
     finally {
-      cleanUp(taskToolbox, failure);
+      cleanUp(taskToolbox, taskStatus);
     }
   }
 
   public abstract TaskStatus runTask(TaskToolbox taskToolbox) throws Exception;
 
-  public void cleanUp(TaskToolbox toolbox, boolean failure) throws Exception
+  public void cleanUp(TaskToolbox toolbox, TaskStatus taskStatus) throws Exception
   {
-    if (toolbox.getConfig().isEncapsulatedTask()) {
-      // report back to the overlord
-      UpdateStatusAction status = new UpdateStatusAction("successful");
-      if (failure) {
-        status = new UpdateStatusAction("failure");
-      }
-      toolbox.getTaskActionClient().submit(status);
-      toolbox.getTaskActionClient().submit(new UpdateLocationAction(TaskLocation.unknown()));
-
-      if (reportsFile != null && reportsFile.exists()) {
-        toolbox.getTaskLogPusher().pushTaskReports(id, reportsFile);
-        log.debug("Pushed task reports");
-      } else {
-        log.debug("No task reports file exists to push");
-      }
-    } else {
+    if (!toolbox.getConfig().isEncapsulatedTask()) {
```

Comment:
@churromorales @nlippis - does the `druid.indexer.task.encapsulatedTask` has to be a user-facing setting? seems like we could just set it to `true` always if the task runner is k8s.

Reply:
just to clarify, it's not a blocker comment for this PR.

Reply:
We could, especially since the peon depends on the `druid.indexer.runner.type` setting.